src/Components/Web.JS/src/Services/ComponentDescriptorDiscovery.ts

```diff
@@ -26,6 +26,35 @@ function discoverServerComponents(document: Document): ServerComponentDescriptor
   return discoveredComponents.sort((a, b): number => a.sequence - b.sequence);
 }
 
+const blazorStateCommentRegularExpression = /\W*Blazor-Component-State:(?<state>[a-zA-Z0-9\+\/=]+)$/;
+
+export function discoverPersistedState(node: Node): string {
+  if (node.nodeType === Node.COMMENT_NODE) {
+    const content = node.textContent || '';
+    const parsedState = blazorStateCommentRegularExpression.exec(content);
+    const value = parsedState && parsedState.groups && parsedState.groups['state'];
+    if(value){
+      node.parentNode?.removeChild(node);
+    }
+    return value || '';
+  }
+
+  if (!node.hasChildNodes()) {
+    return '';
+  }
+
+  const nodes = node.childNodes;
+  for (let index = 0; index < nodes.length; index++) {
+    const candidate = nodes[index];
+    const result = discoverPersistedState(candidate);
+    if(result && result !== ''){
```

Comment:
An empty string is falsey, so the `result !== ''` part is redundant. In any case if we change this to use `null` instead of empty string for "no match found" this can still be just `if (result)`.